internal/registry/registry.go

```diff
@@ -224,22 +227,31 @@ func (r *registry) getLayersDigestsFromManifestStream(manifestStream []byte) ([]
 }
 
 func (r *registry) GetHeaderDataFromLayer(layer v1.Layer, headerName string) ([]byte, error) {
-
-	targz, err := layer.Compressed()
+	reader, readerCloser, err := r.getHeaderReaderFromLayer(layer, headerName)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get targz from layer: %w", err)
+		return nil, fmt.Errorf("failed to get reader for the header %s in the layer: %v", headerName, err)
 	}
-	// err ignored because we're only reading
-	defer targz.Close()
+	defer readerCloser.Close()
+	buff, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read header %s data: %v", headerName, err)
+	}
+	return buff, nil
+}
 
-	gr, err := gzip.NewReader(targz)
+func (r *registry) getHeaderReaderFromLayer(layer v1.Layer, headerName string) (io.Reader, io.ReadCloser, error) {
```

Comment:
```suggestion
func (r *registry) getHeaderReaderFromLayer(layer v1.Layer, headerName string) (io.Reader, io.Closer, error) {
```
I guess we don't want callers to be able to keep reading the layer.

Reply:
I think providing this method with a callback function is safer solution. That way we can close everything we need ourselves.

Reply:
This is an inner function of the registry package. User does not need to know what and how we are doing inside. Currently there are 2 methods that are using it: one is GetHeaderDataFromLayer, which calls this method, and then uses the io.Reader to read the data, closes  the io.ReadCloser and return the data to user. The second one is VerifyModuleExists, which calls this method, and then just closes the io.ReadCloser, since it only wants to verify that the header exists. I think that providing a callback will only complicate the flow